src/components/SourcesTree.js

```diff
@@ -38,6 +58,32 @@ let SourcesTree = React.createClass({
   },
 
   componentWillReceiveProps(nextProps) {
+    if (isEnabled("showSource") &&
+    nextProps.shownSource != this.props.shownSource) {
+      const addToItemsList = this.props.addToItemsList;
+      const itemsList = this.props.itemsList;
+      const sourceURL = nextProps.shownSource;
+      const sourceTreeList = this.state.sourceTree;
+      const itemsStrings = returnItemsStrings(sourceURL);
+      const processIndex = 0;
+
+      // Determine which item(s) should be added to itemsList
+      while (processIndex < itemsStrings.length) {
+        const i = 0;
+        const found = false;
+        while (!found) {
+          if (itemsStrings[processIndex] == sourceTreeList.contents[i].path) {
+            addToItemsList(sourceTreeList.contents[i], true);
```

Comment:
hmm, I think you can use react `setState` here instead of redux.

Also, maybe set the items at the end as opposed to one at a time